src/functions/services/logging/logger.ts

```diff
@@ -0,0 +1,39 @@
+import { Context } from "aws-lambda";
+import { LogMessage, RegisteredLogMessages } from "./types";
+import { LogAttributes } from "@aws-lambda-powertools/logger/lib/cjs/types/Log";
+
+export interface ILogger<T extends string> {
+  log(message: T, data: LogAttributes): void;
+  addContext(lambdaContext: Context): void;
+  appendKeys: (keys: { authSessionId: string }) => void;
+}
+
+export interface ILoggerAdapter<T extends string> {
+  info: (message: LogMessage<T>, data: LogAttributes) => void;
+  addContext: (lambdaContext: Context) => void;
+  appendKeys: (keys: { authSessionId: string }) => void;
+}
+
+export class Logger<T extends string> implements ILogger<T> {
+  constructor(
+    private logger: ILoggerAdapter<T>,
+    private registeredLogs: RegisteredLogMessages<T>,
+  ) {}
+  log = (message: T, data: LogAttributes = {}): void => {
+    this.logger.info(
+      {
+        message,
+        messageCode: this.registeredLogs[message].messageCode,
+      },
+      data,
+    );
+  };
+
+  addContext = (context: Context): void => {
```

Comment:
Is logging `context` a nice to have or has `context` has ever/will ever provide us with useful information?

I'm not suggesting we remove it just keen to know more about it relevant to this project.

Reply:
It adds this which I think is very useful:
<img width="1023" alt="image" src="https://github.com/govuk-one-login/mobile-id-check-async/assets/115020902/c39f33da-9bac-4bad-a320-780463e0b11a">